src/Stack/Package.hs

```diff
@@ -1106,54 +1108,33 @@ getDependencies component dotCabalPath =
 parseDumpHI
     :: FilePath -> RIO Ctx (Set ModuleName, [Path Abs File])
 parseDumpHI dumpHIPath = do
-    dir <- asks (parent . ctxFile)
-    dumpHI <- liftIO $ filterDumpHi <$> fmap CL8.lines (CL8.readFile dumpHIPath)
-    let startModuleDeps =
-            dropWhile (not . ("module dependencies:" `CL8.isPrefixOf`)) dumpHI
-        moduleDeps =
-            S.fromList $
-            mapMaybe (D.simpleParse . TL.unpack . TLE.decodeUtf8) $
-            CL8.words $
-            CL8.concat $
-            CL8.dropWhile (/= ' ') (fromMaybe "" $ listToMaybe startModuleDeps) :
-            takeWhile (" " `CL8.isPrefixOf`) (drop 1 startModuleDeps)
-        thDeps =
-            -- The dependent file path is surrounded by quotes but is not escaped.
-            -- It can be an absolute or relative path.
-                  TL.unpack .
-                  -- Starting with GHC 8.4.3, there's a hash following
-                  -- the path. See
-                  -- https://github.com/yesodweb/yesod/issues/1551
-                  TLE.decodeUtf8 .
-                  CL8.takeWhile (/= '\"') <$>
-            mapMaybe (CL8.stripPrefix "addDependentFile \"") dumpHI
-    thDepsResolved <- liftM catMaybes $ forM thDeps $ \x -> do
-        mresolved <- liftIO (forgivingAbsence (resolveFile dir x)) >>= rejectMissingFile
-        when (isNothing mresolved) $
+  dir <- asks (parent . ctxFile)
+  content <- liftIO $ CL8.readFile dumpHIPath
```

Comment:
I'd rather not using lazy I/O here at all. _If_ we are going to use lazy I/O, I'd much rather use the `withBinaryFile` approach which ensures that the handle is closed at the end. But doesn't the `binary` package provide a streaming parse interface?

I realize this isn't the way the original code was written...

Reply:
By `streaming interface` are you talking about [the pipes-binary package?](http://hackage.haskell.org/package/pipes-binary-0.4.2) If so, do you want me to use it here?

Reply:
I meant a function like [`decodeFileOrFail`](https://www.stackage.org/haddock/lts-13.7/binary-0.8.6.0/Data-Binary.html#v:decodeFileOrFail).

Reply:
Ok let's use that